src/test/java/cloud/fogbow/ras/core/OrderControllerTest.java

```diff
@@ -167,36 +146,271 @@ public void testGetInactiveOrder() throws InstanceNotFoundException {
     @Test(expected = InstanceNotFoundException.class)
     public void testGetInvalidOrder() throws FogbowException {
         // exercise
-        this.ordersController.getOrder("invalid-order-id");
+        this.ordersController.getOrder(INVALID_ORDER_ID);
+    }
+
+    //test case: Checks if attempt to activate null order throws UnexpectedException
+    @Test(expected = UnexpectedException.class)
+    public void testActivateNullOrder() throws FogbowException {
+        // set up
+        Order order = null;
+
+        // exercise
+        this.ordersController.activateOrder(order);
+    }
+
+    // test case: Attempt to activate the same order more than
+    // once must throw UnexpectedException
+    @Test(expected = UnexpectedException.class)
+    public void testActivateOrderTwice() throws FogbowException {
+        // set up
+        Order order = createLocalOrder(getLocalMemberId());
+
+        // exercise
+        this.ordersController.activateOrder(order);
+        this.ordersController.activateOrder(order);
+    }
+
+    // test case: Attempt to deactivate the same order more than
+    // once must throw UnexpectedException
+    @Test(expected = UnexpectedException.class)
+    public void testDeactivateOrderTwice() throws FogbowException {
+        // set up
+        Order order = createLocalOrder(getLocalMemberId());
+
+        // exercise
+        this.ordersController.activateOrder(order);
+        this.ordersController.deactivateOrder(order);
+        this.ordersController.deactivateOrder(order);
+    }
+
+    // test case: Checks if deactivateOrder changes it's state to DEACTIVATED
+    @Test
+    public void testDeactivateOrderSuccess() throws FogbowException {
+        // set up
+        Order order = createLocalOrder(getLocalMemberId());
+
+        // exercise
+        this.ordersController.activateOrder(order);
+        this.ordersController.deactivateOrder(order);
+
+        // verify
```

Comment:
i think we should verify before the exercise too

Reply:
Okay, consider as done.